registry/extension/oras/referrers.go

```diff
@@ -41,6 +58,37 @@ func (h *referrersHandler) getReferrers(w http.ResponseWriter, r *http.Request)
 		referrers = []orasartifacts.Descriptor{}
 	}
 
+	// only consider pagination if # of referrers is greater than page size
+	if len(referrers) > nPage {
+		startIndex := 0
+		if nextToken != "" {
+			for i, ref := range referrers {
+				if ref.Digest.String() == nextToken {
+					startIndex = i + 1
+				}
+			}
+			// if matching referrer not found
```

Comment:
The referrer manifest found previously might be deleted now, so one option is to add a few digests instead of just one to the nextToken.